OData/test/UnitTest/System.Web.OData.Test/OData/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -88,6 +89,41 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForSing
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData(false, "http://localhost/MyOrders(42)/OrderLines(21)/OrderLines")]
```

Comment:
Yes, existing test case `GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForEntitySet` works fine.
```c#
         [InlineData(false, "http://localhost/Customers(42)/Orders")]
         [InlineData(true, "http://localhost/Customers(42)/NS.SpecialCustomer/Orders")]
```

Feel free to change the model `CustomersModelWithInheritance`.